lib/Compiler.js

```diff
@@ -59,13 +64,29 @@ function compileSolc(logger, contractFiles, cb) {
           compiled_object[className].functionHashes = contract.hashes;
           compiled_object[className].abiDefinition = JSON.parse(contract.abi);
           compiled_object[className].filename = fileName;
+
+          
         }
 
         fileCb();
       });
     },
     function (err) {
-      cb(err, compiled_object);
+      cb(err, compiled_object);  
+      
+      embark.events.on("outputDone", function() {
+        if(outputBinary){
```

Comment:
Should put the if on the outside, because otherwise the event listener is useless